persistence-modules/read-only-transactions/src/main/java/com/baeldung/read_only_transactions/mysql/spring/ReadOnlyContext.java

```diff
@@ -0,0 +1,22 @@
+package com.baeldung.read_only_transactions.mysql.spring;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ReadOnlyContext {
+
+    private static final ThreadLocal<AtomicInteger> READ_ONLY_LEVEL = ThreadLocal.withInitial(() -> new AtomicInteger(0));
+
+    private ReadOnlyContext() {}
+
+    public static boolean isReadOnly() {
+        return READ_ONLY_LEVEL.get().get() > 0;
+    }
+
+    public static void enter() {
+        READ_ONLY_LEVEL.get().incrementAndGet();
+    }
+
+    public static void exit() {
+        READ_ONLY_LEVEL.get().decrementAndGet();
```

Comment:
Once you exit, shouldn´t you clear the value, that is, calling the ThreadLocal remove method?

Reply:
No, because you may have nested transactions.